nagplug/__init__.py

```diff
@@ -366,7 +366,8 @@ def add_extdata(self, message):
         arguments:
             message: a free-form string
         """
-        self._extdata.append(str(message))
+        message = str(message).replace('|', '!')
+        self._extdata.append()
```

Comment:
You still need to append the modified message

Reply:
Thanks, I fixed it.